app/controllers/grade_entry_forms_controller.rb

```diff
@@ -440,7 +479,29 @@ def csv_upload
         flash[:error] = I18n.t('csv.upload.non_text_file_with_csv_extension')
       end
     end
-    redirect_to action: 'grades', id: @grade_entry_form.id
+    if !matching_names && params[:id].to_s != @grade_entry_form.id.to_s
+      redirect_to csv_overwrite_grade_entry_form_path(params[:id], with: @grade_entry_form.id)
+    else
+      redirect_to action: 'grades', id: @grade_entry_form.id
+    end
+  end
+
+  def csv_overwrite
```

Comment:
Does it complain that the function is too complex? Is there any need to simplify it/suggestions?